internal/extsvc/bitbucketcloud/client.go

```diff
@@ -245,11 +244,12 @@ func (c *client) do(ctx context.Context, req *http.Request, result any) error {
 	sleepTime := 10 * time.Second
 	for {
 		resp, err = oauthutil.DoRequest(ctx, nil, c.httpClient, req, c.Auth)
+		code = resp.StatusCode
```

Comment:
It looks like `resp` is an `*http.Response` here, so we probably need to make sure `resp` isn't `nil` before assigning `code = resp.StatusCode`, or else you could get more invalid memory address runtime panics:

```suggestion
		resp, err = oauthutil.DoRequest(ctx, nil, c.httpClient, req, c.Auth)
		if resp != nil {
			code = resp.StatusCode
		}
```